### PR TITLE
ENH: adopt sanitize_value from dandi-cli and use for sanitization of identifier

### LINKED ISSUE #172: Incorrect summaries (at least number of subjects) in some dandisets
It was mentioned by the dandiset author for https://dandiarchive.org/dandiset/000212 that there is only 1013 files but summary has 1097 subjects!

I did check across all dandisets for which we have datalad representations:

```shell
(dandisets) dandi@drogon:/mnt/backup/dandi/dandisets$ for d in 0*; do ( cd $d; n1=$(awk '/numberOfSubjects:/{print $2}' dandiset.yaml || echo 0 ); n2=$(/bin/ls -1d sub-* 2>/dev/null | wc -l || echo 0); [ -n "$n1" ] && { [ "$n1" = "$n2" ] || echo "`pwd`: Mismatch $n1 != $n2"; } ); done
/mnt/backup/dandi/dandisets/000025: Mismatch 1 != 0
/mnt/backup/dandi/dandisets/000043: Mismatch 22 != 13
/mnt/backup/dandi/dandisets/000051: Mismatch 1 != 0
/mnt/backup/dandi/dandisets/000052: Mismatch 1 != 0
/mnt/backup/dandi/dandisets/000058: Mismatch 1 != 0
/mnt/backup/dandi/dandisets/000168: Mismatch 30 != 0
/mnt/backup/dandi/dandisets/000212: Mismatch 1097 != 1013
/mnt/backup/dandi/dandisets/000228: Mismatch 64 != 32
/mnt/backup/dandi/dandisets/000246: Mismatch 12 != 11
/mnt/backup/dandi/dandisets/000288: Mismatch 12 != 9
/mnt/backup/dandi/dandisets/000293: Mismatch 201 != 121
/mnt/backup/dandi/dandisets/000297: Mismatch 197 != 118
/mnt/backup/dandi/dandisets/000341: Mismatch 310 != 0
/mnt/backup/dandi/dandisets/000362: Mismatch 11 != 0
/mnt/backup/dandi/dandisets/000454: Mismatch 5 != 4
```

so it seems that the issue manifests itself quite wildly.

### PR DESCRIPTION
**Pull request by yarikoptic**
Closes #172 as empirically shown 

@satra do you see a quick way to add a test?

TODO
- [x] compare to current (time has passed; 0.63.0-2-g5faa18a4) normalization in dandi-cli
- [ ] after release - adjust dandi-cli to use this normalization implementation

### CODE REVIEW

**Comment by satra** [comment]
you could use an asset in the tests directory and change the "path" key and subject identifier keys to make the aggregate function raise that assertion you put.

**Comment by yarikoptic** [comment]
I didn't add assertion in this PR. In this PR such assertion would not be appropriate since code would still allow for multiple subjects from a single path.

**Comment by satra** [comment]
you could manually assert in the test.

**Comment by yarikoptic** [comment]
> you could manually assert in the test.

well, all tests assert "manually", the question what and in which test.... well -- since confirmed empirically, just feel free to either merge as is, or extend with the test as you see fit.

**Comment by satra** [comment]
@yarikoptic - this needs to be updated and merged for #172

File: dandischema/utils.py [hunk 1]
```diff
@@ -114,3 +114,25 @@ def strip_top_level_optional(type_: Any) -> Any:
...
+
+    When `field` is not "extension", we also replace ".".
+
+    Based on dandi.organize._sanitize_value.
+
+    .. versionchanged:: 0.8.3
+
+            ``sanitize_value`` added
+    """
+    value = re.sub(r"[_*\\/<>:|\"'?%@;]", sub, value)
```
Comment (yarikoptic) [inline dandischema/utils.py:135]: in dandi-cli ATM

```
dandi/organize.py=def _sanitize_value(value, field):
dandi/organize.py:    value = re.sub(r"[_*\\/<>:|\"'?%@;,\s]", "-", value)
```

so let's add those new (? added in Nov 2023 in 8d7b1a6492a8e51875285f89afce08fd2911281d so after initiated this PR)

```suggestion
    value = re.sub(r"[_*\\/<>:|\"'?%@;,\s]", sub, value)
```

**Pull request merged**